### PR TITLE
Merge release 1.15.1-hf1 to release/1.15.2 and update common dependency and submodule to point to latest common-0.0.7

### PR DESCRIPTION
**Pull request by kreedula**

### CODE REVIEW

File: adal/build.gradle [hunk 1]
```diff
@@ -135,10 +135,10 @@ dependencies {
         transitive = false
     }
 
-    snapshotApi(group: 'com.microsoft.identity', name: 'common', version: '0.0.3-SNAPSHOT', changing: true)
+    snapshotApi(group: 'com.microsoft.identity', name: 'common', version: '0.0.7', changing: true)
```
Comment (iambmelt) [inline adal/build.gradle]: 👍

**Comment by iambmelt** [comment]
Did the ADAL copy of `HashMapExtensions` get removed? I don't see it being deleted but I also don't see that it exists on this branch.... 🤔

**Comment by kreedula** [comment]
> Did the ADAL copy of `HashMapExtensions` get removed? I don't see it being deleted but I also don't see that it exists on this branch.... 🤔

Yes It's deleted, for some weird  reason deletion its not showing on the PR and imports are updated too.

**Comment by iambmelt** [comment]
> > Did the ADAL copy of `HashMapExtensions` get removed? I don't see it being deleted but I also don't see that it exists on this branch.... 🤔
> 
> Yes It's deleted, for some weird reason deletion its not showing on the PR and imports are updated too.

In that case, approving. Looks good (well, what I can see, anyway!)

Thanks @kreedula

**Comment by kreedula** [comment]
Thanks @iambmelt

**Pull request merged**